r/rsparkling/R/package.R

```diff
@@ -1,13 +1,24 @@
-
 #' @import sparklyr h2o utils
 NULL
 
 # define required spark packages
 spark_dependencies <- function(spark_version, scala_version, ...) {
+  sw_version = getOption("rsparkling.sparklingwater.version", default = "1.6.7")
+
+  if(as.package_version(spark_version)$major != as.package_version(sw_version)$major){
+    stop(cat(paste0("Major version of Sparkling Water does not correspond to major Spark version.
+    \nMajor Sparkling Water Version = ",as.package_version(sw_version)$major,
+    "\nMajor Spark Version = ",as.package_version(spark_version)$major)))
+  }
+  if(as.package_version(spark_version)$minor != as.package_version(sw_version)$minor){
```

Comment:
I think that we shouldn't stop the application if the minor versions differ, I would just warn in that case. We should stop only in case when major version differ.